pluton/spawn/containercache/sshcache.go

```diff
@@ -0,0 +1,278 @@
+package containercache
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/coreos/mantle/platform"
+	"github.com/coreos/mantle/util"
+)
+
+const (
+	cacheDir          = "/home/core/containercache"
+	downloadRetries   = 3
+	downloadRetryWait = 15 * time.Second
+	downloadTimeout   = 2 * time.Minute
+)
+
+var once sync.Once
+
+type ImageName struct {
+	Name   string
+	Engine string // docker or rkt
+}
+
+// StartBastionOnce will populate a bastion machine with the given containers
+// for rkt and docker. Since its best to run this only once per test suite, the
+// bastion machine should be in a separate platform.Cluster then the one used
+// in the test so it can be available for all tests in a given run. This
+// function is guarded with a sync.Once, so only the first within a test suite
+// it will have it run.
+func StartBastionOnce(bastion platform.Machine, names []ImageName) error {
```

Comment:
It may be helpful to drop this run once magic and explicitly manage the life cycle of the bastion in the top level main() function or similar.